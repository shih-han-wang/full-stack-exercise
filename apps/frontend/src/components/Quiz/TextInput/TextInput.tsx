import { Input } from '@chakra-ui/react'

type Props = {
  onChange: () => void
  value: string
  disabled?: boolean
}

const TextInput = ({ onChange, value, disabled = false }: Props) => (
  <Input
    disabled={disabled}
    onChange={onChange}
    placeholder='Answer...'
    size='lg'
    value={value}
    width='320px'
  />
)

export default TextInput
