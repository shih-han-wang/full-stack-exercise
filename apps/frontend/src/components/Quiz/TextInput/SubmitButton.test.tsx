import { render } from '@testing-library/react'
import TextInput from './TextInput'

const onChangeMock = jest.fn()

describe('Components > Quiz > TextInput', () => {
  it('should render correctly', async () => {
    const view = render(<TextInput value='value' onChange={onChangeMock} />)

    expect(view.asFragment()).toMatchSnapshot()
  })

  it('should render disabled state correctly', async () => {
    const view = render(
      <TextInput value='value' onChange={onChangeMock} disabled />
    )

    expect(view.asFragment()).toMatchSnapshot()
  })
})
