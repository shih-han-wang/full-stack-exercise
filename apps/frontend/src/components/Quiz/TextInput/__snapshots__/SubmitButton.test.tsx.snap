// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Quiz > TextInput should render correctly 1`] = `
<DocumentFragment>
  <input
    class="chakra-input css-0"
    placeholder="Answer..."
    value="value"
  />
</DocumentFragment>
`;

exports[`Components > Quiz > TextInput should render disabled state correctly 1`] = `
<DocumentFragment>
  <input
    class="chakra-input css-0"
    disabled=""
    placeholder="Answer..."
    value="value"
  />
</DocumentFragment>
`;
